docs/api.rst

```diff
@@ -0,0 +1,45 @@
+API
+===
+
+.. toctree::
+   :maxdepth: 2
+
+.. _api-tracing:
+
+Tracing
+^^^^^^^
+
+.. py:module:: ddtrace
+
+.. autofunction:: ddtrace.patch_all
+
+.. autofunction:: ddtrace.patch
+
+.. autoclass:: ddtrace.Tracer
+    :members:
+
+.. autoclass:: ddtrace.Span
+    :members:
+
+.. autoclass:: ddtrace.Pin
+    :members:
+
+.. autoclass:: ddtrace.context.Context
```

Comment:
We should probably re-export `Context` in `ddtrace/__init__.py` so we have `ddtrace.Context` here.

Otherwise we might want to document the `ddtrace.context` module separately from `ddtrace` module. (not sure what I mean by this/how it should be implemented)

Reply:
I wonder if maybe we should have:

```
# API
  ## Tracing
    ### ddtrace
    ### ddtrace.context
    ### ddtrace.sampler
    ### ddtrace.propagation.http
```

And put the modules as headings to separate them ?

Reply:
> We should probably re-export `Context` in `ddtrace/__init__.py` so we have `ddtrace.Context` here.

Hm, this does make sense to add it to `ddtrace.__all__`. Should we do the same with the other classes?

- ddtrace.sampler.DatadogSampler
- ddtrace.sampler.SamplingRule
- ddtrace.propagation.http.HTTPPropagator

@mabdinur @Kyle-Verhoog

Reply:
We can handle as a follow-up.